packages/wallets/src/components/Base/WalletPasswordField/WalletPasswordField.tsx

```diff
@@ -1,40 +1,46 @@
-import React, { useMemo, useState } from 'react';
+/* eslint-disable sort-keys */
+import React, { useState } from 'react';
 import { zxcvbn } from '@zxcvbn-ts/core';
+import { Score } from '@zxcvbn-ts/core/dist/types';
 import { WalletTextField } from '../WalletTextField';
 import { WalletTextFieldProps } from '../WalletTextField/WalletTextField';
 import PasswordMeter, { PasswordMeterProps } from './PasswordMeter';
 import PasswordViewerIcon from './PasswordViewerIcon';
 import './WalletPasswordField.scss';
 
-type StrengthMessage = Record<1 | 2 | 3 | 4, string>;
-
 interface WalletPasswordFieldProps extends WalletTextFieldProps, PasswordMeterProps {
-    messageObj?: StrengthMessage;
+    password: string;
     showPasswordMeter?: boolean;
 }
 
-const WalletPasswordField: React.FC<WalletPasswordFieldProps> = ({ messageObj, showPasswordMeter = true }) => {
-    const [password, setPassword] = useState('');
+const WalletPasswordField: React.FC<WalletPasswordFieldProps> = ({ password, showPasswordMeter = true, ...rest }) => {
     const [viewPassword, setViewPassword] = useState(false);
-    const hasMessage = !!messageObj;
 
-    const passwordStrength = zxcvbn(password).score;
-    const progressText = useMemo(() => {
-        return messageObj ? messageObj[passwordStrength as keyof StrengthMessage] : '';
-    }, [messageObj, passwordStrength]);
+    const passwordScore = password ? zxcvbn(password).score : undefined;
+
+    const helperMessage: Partial<Record<Score, string>> = {
+        0: 'You should enter 8 - 25 characters',
+        1: 'Password should have lower and uppercase English letters with numbers.',
+        2: 'This is a very common password',
+        3: '',
+        4: '',
+    };
 
     return (
         <div className='wallets-password'>
             <WalletTextField
-                helperMessage={progressText}
-                onChange={e => setPassword(e.target.value)}
+                helperMessage={helperMessage[passwordScore ?? 0]}
+                inputClassName='wallets-password__input'
+                pattern='^(?=.*[a-z])(?=.*\d)(?=.*[A-Z])[!-~]{8,25}'
                 renderRightIcon={() => (
                     <PasswordViewerIcon setViewPassword={setViewPassword} viewPassword={viewPassword} />
                 )}
-                showMessage={hasMessage}
+                showMessage={true}
                 type={viewPassword ? 'text' : 'password'}
+                value={password}
+                {...rest}
```

Comment:
not sure if ...rest is good pattern, it might cause accidental/undesired properties to be passed on, 
generally, passing things explicitly listed is preferred, 

If I add prop specific to given compnent, I do not expect it to be passed multiple layers down the line to components which might or might not support that prop or might have undesired side effects,

Reply:
I am passing down the `WalletTextField` props because I am still using it in the `WalletPasswordField`.

Reply:
Yes, they are used, but they should be passed in explicit manner. You do not want the child component to break just beacuse someone added property in the component two layers higher in the structure. I think we've already had comments about not using ...rest for that purspose in one of the PRs, just cant find them now

Reply:
e.g. instead of ...rest, just pass the specific props which are needed and are supposed to be passed (easier to read and less prone to bugs and works with intellisense/typescript, unlike ...rest which can cover pretty much anything - its simliar to usage of "any" as anything can be there and won't be caught up by typescript)

Reply:
Understood, I already made my changes and only included the props that are needed for the password field 🙏🏼